File: Ruby/Problem2.rb
def main
    sum = 2
    a1 = 1
    a2 = 2
    temp = 0
    while a1 + a2 < 4000000
        temp = a1 + a2
        if (temp) % 2 == 0
            sum += temp
        end
        a1, a2 = a2, temp
    end
    puts sum
end

if __FILE__ == $0
    main
end

File: C/Problem2.c
#include <stdlib.h>
#include <stdio.h>

#define MAX_FIB 4000000

int main() {
    int prev_fib = 0;
    int current_fib = 1;
    int temp;
    int sum = 0;

    while( current_fib < MAX_FIB ) {
        if ( current_fib % 2 == 0 ) {
            sum += current_fib;
        }
        temp = current_fib;
        current_fib = prev_fib + current_fib;
        prev_fib = temp;
    }

    printf( "The sum of even Fib nums < 4000000 is %d\n", sum );
}
